### PR TITLE
Add Doxygen tool for compiling C++ documentation

### PR DESCRIPTION
**Pull request by ctrlaltdavid**
See /tools/doxygen/README.md for usage instructions.

- It is configured to generate documentation only for items that actually have documentation. (There is a little documentation in the codebase already.)
- This C++ documentation is separate from the scripting API documentation, which is also generated from the C++ code using JSDoc.
- In order to prevent the C++ and scripting API documentation source from clashing, all occurrences of the `/**jsdoc` JSDoc tag have been changed to `/*@jsdoc`.

### CODE REVIEW

File: tools/doxygen/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+## Altering the Doxygen configuration
+
+This is most easily accomplished by using the Doxyfile GUI and using it to write a new `Doxyfile` file.
+Alternatively, you can edit the `Doxyfile` configuration file in a text editor.
+
+
+## References
+
+- [Doxygen Web Site](https://www.doxygen.nl/)
```
Comment (digisomni) [inline tools/doxygen/README.md:54]:
```suggestion
- [Doxygen Website](https://www.doxygen.nl/)
```

File: tools/doxygen/Doxyfile [hunk 2]
```diff
@@ -0,0 +1,2665 @@
...
+# quick idea about the purpose of the project. Keep the description short.
+
+PROJECT_BRIEF          =
+
+# With the PROJECT_LOGO tag one can specify a logo or an icon that is included
+# in the documentation. The maximum height of the logo should not exceed 55
+# pixels and the maximum width should not exceed 200 pixels. Doxygen will copy
+# the logo to the output directory.
+
+PROJECT_LOGO           = D:/Workspaces/CtrlAltStudio/Vircadia-A/Vircadia-A/tools/doxygen/vircadia.png
```
Comment (digisomni) [inline tools/doxygen/Doxyfile:54]:
```suggestion
PROJECT_LOGO           = vircadia.png
```

**Review by digisomni — CHANGES_REQUESTED** [review]

**Review by digisomni — APPROVED** [review]

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, client)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1212/Vircadia-Interface-PR1212-dfea9d8-_f3ee99fe.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1212/Vircadia-PR1212-dfea9d8-.tar_f3ee99fe.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1212/Vircadia-PR1212-dfea9d8-_f3ee99fe.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1212/Vircadia-PR1212-dfea9d8-.tar_f3ee99fe.gz

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1212/Vircadia-PR1212-dfea9d8-_f3ee99fe.dmg

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1212/Vircadia-PR1212-7e31bd7-_27779030.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1212/Vircadia-PR1212-7e31bd7-.tar_27779030.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1212/Vircadia-PR1212-7e31bd7-.tar_27779030.gz

**build (macOS-latest, client)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1212/Vircadia-Interface-PR1212-7e31bd7-_27779030.dmg

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1212/Vircadia-PR1212-7e31bd7-_27779030.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1212/Vircadia-PR1212-7e31bd7-_27779030.exe

**Pull request merged**